src/nsrt_learning.py

```diff
@@ -75,13 +77,25 @@ def segment_trajectory(trajectory: GroundAtomTrajectory) -> List[Segment]:
     for t in range(len(actions)):
         current_segment_traj[0].append(states[t])
         current_segment_traj[1].append(actions[t])
-        if all_atoms[t] != all_atoms[t+1]:
+        switch = all_atoms[t] != all_atoms[t+1]
+        if actions[t].has_option() and t < len(actions) - 1:
+            option_t = actions[t].get_option()
+            option_t1 = actions[t+1].get_option()
+            option_t_spec = (option_t.parent, option_t.objects)
+            option_t1_spec = (option_t1.parent, option_t1.objects)
+            if option_t_spec != option_t1_spec:
+                switch = True
+        if switch:
             # Include the final state as both the end of this segment
             # and the start of the next segment.
             # Include the default option here; replaced during option learning.
```

Comment:
This comment can be updated since it's not true in the non-option-learning code path.